File: 2015 czerwiec/c++/3.cpp
#include <iostream>
#include <fstream>
#include <string>
using namespace std;

string int_to_kresk(int r)
{
    switch (r)
    {
        case 0:
            return "10101110111010";
        case 1:
            return "11101010101110";
        case 2:
            return "10111010101110";
        case 3:
            return "11101110101010";
        case 4:
            return "10101110101110";
        case 5:
            return "11101011101010";
        case 6:
            return "10111011101010";
        case 7:
            return "10101011101110";
        case 8:
            return "11101010111010";
        case 9:
            return "10111010111010";
    }
    return "";
}

int main()
{
    ifstream in("kody.txt");
    ofstream out("kody3.txt");

    for (int linia = 1; linia <= 500; linia++)
    {
        out << "11011010";
        string s;
        in >> s;
        int even = 0, odd = 0;
        for (int i = 0; i < s.size(); i++)
        {
            out << int_to_kresk(s[i] - '0');
            if ((i + 1) % 2)
                odd += s[i] - '0';
            else
                even += s[i] - '0';
        }
        int r = (10 - (even * 3 + odd) % 10) % 10;
        out << int_to_kresk(r) << '\n';
    }    
}

File: 2015 czerwiec/python/3.py
with open("kody.txt", "r") as f:
    k = f.read().strip().split('\n')

kody = {0: '10101110111010', 1: '11101010101110', 2: '10111010101110', 3: '11101110101010', 4: '10101110101110', 5: '11101011101010', 6: '10111011101010', 7: '10101011101110', 8: '11101010111010', 9: '10111010111010'}

with open("kody3.txt", "w") as f:
    for liczba in k:
        even, odd = 0, 0
        f.write('11011010')
        for i in range(len(liczba)):
            f.write(kody[int(liczba[i])])
            if (i + 1) % 2:
                odd += int(liczba[i])
            else:
                even += int(liczba[i])
        r = (10 - ((even * 3 + odd) % 10)) % 10
        f.write(kody[r] + '\n')